callbacks.go

```diff
@@ -117,15 +104,15 @@ func (exec *sidecarExecutor) LaunchTask(driver executor.ExecutorDriver, taskInfo
 
 // KillTask is a Mesos callback that will try very hard to kill off a running
 // task/container.
-func (exec *sidecarExecutor) KillTask(driver executor.ExecutorDriver, taskID *mesos.TaskID) {
-	log.Infof("Killing task: %s", *taskID.Value)
-
-	// Instruct Sidecar to set the status of the service to DRAINING
-	exec.notifyDrain()
+func (exec *sidecarExecutor) KillTask(taskID *mesos.TaskID) {
+	log.Infof("Killing task: %s", taskID.Value)
 
 	// Stop watching the container so we don't send the wrong task status
 	go func() { exec.watchLooper.Quit() }()
 
+	// Instruct Sidecar to set the status of the service to DRAINING
+	exec.notifyDrain()
+
```

Comment:
In testing, `notifyDrain()` and `exec.watchLooper.Quit()` seemed to be in the wrong order. I am not sure why that worked before.

Reply:
Good catch! Removing the goroutine wrapper. Thanks for reviewing.